lib/puppet/defaults.rb

```diff
@@ -407,7 +407,29 @@ def self.default_diffargs
     :node_terminus => {
       :type       => :terminus,
       :default    => "plain",
-      :desc       => "Where to find information about nodes.",
+      :desc       => <<-'EOT',
+        Which node data plugin to use when compiling node catalogs.
+
+        When Puppet compiles a catalog, it combines two primary sources of info: the main manifest,
+        and a node data plugin (often called a "node terminus," for historical reasons). Node data
+        plugins provide three things for a given node name:
+
+        1. A list of classes to add to that node's catalog (and, optionally, values for their
+           parameters).
+        2. Which Puppet environment the node should use.
+        3. A list of additional top-scope variables to set.
+
+        The three main node data plugins are:
+
+        * `plain` --- Returns no data, so that the main manifest controls all node configuration.
```

Comment:
Is `main manifest` a common term for the manifests in a manifest-driven classification workflow (I assume that's what this refers to)?  What if we went even further with something like `...so that puppet manifests such as site.pp control...` ? Or is that too specific..